common4j/src/main/com/microsoft/identity/common/java/util/StringUtil.java

```diff
@@ -416,4 +416,13 @@ public static String base64Encode(final byte[] bytes) {
         }
         return sb.toString();
     }
+
+    /**
+     * A helper function for validating if the given String is null or empty.
+     */
+    public static void throwIfStringIsNull(@NonNull final String objectName, @Nullable final String object) throws IllegalStateException {
```

Comment:
```suggestion
    public static void throwIfArgumentIsNullOrEmpty(@NonNull final String argumentName, @Nullable final String argument) throws IllegalStateException {
```

Reply:
Also you may want to throw a `NullPointerException` instead

Reply:
7cee1dd4643a39521911892cd529e5ca2bfae6fb

Reply:
Thanks for the feedback @shahzaibj, great suggestions